src/main/java/com/google/sps/servlets/MessageServlet.java

```diff
@@ -51,19 +56,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         // then sorts them by timestamp.
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-        Query messageQuery = new Query("message").addSort("timestamp", SortDirection.ASCENDING);
-        PreparedQuery results = datastore.prepare(messageQuery);
+        Query chatroomQuery = new Query("message").addSort("timestamp", SortDirection.ASCENDING);
+        chatroomQuery.setFilter(
+                new CompositeFilter(
+                        CompositeFilterOperator.OR,
+                        Arrays.asList(
+                                new FilterPredicate("senderId", FilterOperator.EQUAL, userID),
+                                new FilterPredicate("recipientId", FilterOperator.EQUAL, userID))));
+        PreparedQuery messages =
+                DatastoreServiceFactory.getDatastoreService().prepare(chatroomQuery);
```

Comment:
Does this work without issue? I remember an earlier trial run for this type of querying produced errors in the servlet

Reply:
Yea it compiled and ran and passed the testers.

Reply:
Did you try deploying it and testing?